src/retro-c.cpp

```diff
@@ -0,0 +1,476 @@
+#include "retro-c.h"
+
+#include "coreinfo.h"
+#include "data.h"
+#include "emulator.h"
+#include "imageops.h"
+#include "memory.h"
+#include "search.h"
+#include "script.h"
+#include "movie.h"
+#include "movie-bk2.h"
+
+#include <map>
+#include <unordered_map>
+#include <unordered_set>
+
+using namespace Retro;
+
+CEmulator* emulatorCreate(const char* romPath) {
+  Retro::Emulator* emulator = new Retro::Emulator();
+  if (Emulator::isLoaded()) {
+    fprintf(
+      stderr,
+      "Cannot create multiple emulator instances per process, make sure to call "
+      "'env.close()' on each environment before creating a new one.");
+    return nullptr;
+  }
+  if (!emulator->loadRom(romPath)) {
+    fprintf(stderr, "Could not load ROM.");
+    return nullptr;
+  }
+  // The following is necessary because otherwise, you get a segfault 
+  // when you try to get the screen for the first time.
+  emulator->run();
+  return new CEmulator {emulator, 0};
+}
+
+void emulatorDelete(CEmulator* emulator) {
+  delete ((Retro::Emulator*) emulator->emulator);
```

Comment:
`static_cast<Emulator>`.

Reply:
I don't usually write C++ code so I'm curious, why is a static cast preferable to the C-style cast? According to [this](https://en.cppreference.com/w/cpp/language/explicit_cast), in this case they should be equivalent.